core/execution/amm/pool.go

```diff
@@ -539,6 +539,14 @@ func (p *Pool) priceForVolumeAtPosition(volume uint64, side types.Side, pos int6
 	if volume == 0 {
 		panic("cannot calculate price for zero volume trade")
 	}
+	// first check for empty curves
```

Comment:
I don't think this is right, its treating the symptom and not the cause. The panic is there because the AMM has got into an invalid state (the AMM is long, but it has no lower curve and so it should never be possible for it to be long) so we're now just avoid the panics here and leaving bad AMMs.

Also the feature test passes when the fix is removed, so I think we need to take a moment to fully understand whats happening in the market-sim run.